src/Data/Abstract/Path.hs

```diff
@@ -2,26 +2,40 @@ module Data.Abstract.Path
   ( dropRelativePrefix
   , joinPaths
   , stripQuotes
+  , joinPathf
   ) where
 
 import           Data.Text (Text)
 import qualified Data.Text as T
 import           System.FilePath.Posix
+import qualified System.Path as Path
+import System.Path.PartClass (FileDir)
 
 -- | Join two paths a and b. Handles walking up relative directories in b. e.g.
 --
 --     joinPaths "a/b" "../c" == "a/c"
 --     joinPaths "a/b" "./c" == "a/b/c"
 --
 -- Walking beyond the beginning of a just stops when you get to the root of a.
-joinPaths :: FilePath -> FilePath -> FilePath
-joinPaths a b = let bs = splitPath (normalise b)
+joinPathf :: FilePath -> FilePath -> FilePath
+joinPathf a b = let bs = splitPath (normalise b)
```

Comment:
This name is a little opaque; I would call it `joinUntypedPaths` or something, and insert a comment that says to prefer `joinPaths` if humanly possible.

Reply:
Fixed!